File: src/main/java/com/back/domain/member/review/dto/ReviewDto.java
package com.back.domain.member.review.dto;

import com.back.domain.member.review.entity.Review;

import java.time.LocalDateTime;

public record ReviewDto(
        int id,
        LocalDateTime createDate,
        LocalDateTime modifyDate,
        int score,
        String comment,
        int memberId,
        int reviewerId
) {
    public ReviewDto(Review review) {
        this(
                review.getId(),
                review.getCreateDate(),
                review.getModifyDate(),
                review.getStar(),
                review.getComment(),
                review.getMember().getId(),
                review.getReviewerId()
        );
    }
}

File: src/main/java/com/back/domain/member/review/dto/ReviewDto.kt
package com.back.domain.member.review.dto

import com.back.domain.member.review.entity.Review
import java.time.LocalDateTime

data class ReviewDto(
    val id: Int,
    val createDate: LocalDateTime?,
    val modifyDate: LocalDateTime?,
    val score: Int,
    val comment: String?,
    val memberId: Int,
    val reviewerId: Int
) {
    constructor(review: Review) : this(
        review.id,
        review.createDate,
        review.modifyDate,
        review.star,
        review.comment,
        review.member.id,
        review.reviewerId
    )
}
